libraries/AP_GPS/AP_GPS_SBF.cpp

```diff
@@ -72,19 +77,30 @@ AP_GPS_SBF::read(void)
     }
 
     if (gps._auto_config != AP_GPS::GPS_AUTO_CONFIG_DISABLE) {
-        if (_init_blob_index < ARRAY_SIZE(_initialisation_blob)) {
+        if (_init_blob_index < (ARRAY_SIZE(_initialisation_blob))) {
             uint32_t now = AP_HAL::millis();
-            const char *init_str = _initialisation_blob[_init_blob_index];
-
             if (now > _init_blob_time) {
                 if (now > _config_last_ack_time + 2500) {
                     // try to enable input on the GPS port if we have not made progress on configuring it
                     Debug("SBF Sending port enable");
                     port->write((const uint8_t*)_port_enable, strlen(_port_enable));
                     _config_last_ack_time = now;
                 } else {
-                    Debug("SBF sending init string: %s", init_str);
-                    port->write((const uint8_t*)init_str, strlen(init_str));
+                    char *init_str = nullptr;
+                    if (!_validated_initial_sso) {
+                        if (_initial_sso == nullptr) {
+                            UNUSED_RESULT(asprintf(&_initial_sso, "sso, Stream%d, COM%d%s",
```

Comment:
You need to explitly null `_initial_sso` here on failure - `asprintf` leaves it undefined on error.